solr/core/src/java/org/apache/solr/search/Grouping.java

```diff
@@ -603,7 +566,7 @@ protected Integer getNumberOfGroups() {
     protected void populateScoresIfNecessary() throws IOException {
       if (needScores) {
         for (GroupDocs<?> groups : result.groups) {
-          TopFieldCollector.populateScores(groups.scoreDocs, searcher, query);
+          TopFieldCollector.populateScores(groups.scoreDocs(), searcher, query);
```

Comment:
Just leaving a note here from investigating the grouping-related test failures, the filling in of the scores here does not influence the `maxScore` of a group and that seems to be wrong (always 1 possibly?) and so the groups are ordered wrongly.